planner/core/expression_rewriter.go

```diff
@@ -220,6 +220,24 @@ func (er *expressionRewriter) constructBinaryOpFunction(l expression.Expression,
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		if evalexpr, ok := expr1.(*expression.Constant); ok {
+			_, isNull, err1 := evalexpr.EvalInt(er.ctx, chunk.Row{})
+			if err1 != nil {
```

Comment:
if err1 != nil || isNil {
    return expr1, err1
}

We do not need the errors.Trace now.